crates/matrix-sdk-ui/src/timeline/traits.rs

```diff
@@ -141,13 +141,27 @@ impl RoomDataProvider for Room {
     }
 
     async fn read_receipts_for_event(&self, event_id: &EventId) -> IndexMap<OwnedUserId, Receipt> {
-        match self.event_receipts(ReceiptType::Read, ReceiptThread::Unthreaded, event_id).await {
-            Ok(receipts) => receipts.into_iter().collect(),
-            Err(e) => {
-                error!(?event_id, "Failed to get read receipts for event: {e}");
-                IndexMap::new()
-            }
-        }
+        let mut unthreaded_receipts =
+            match self.event_receipts(ReceiptType::Read, ReceiptThread::Unthreaded, event_id).await
+            {
+                Ok(receipts) => receipts.into_iter().collect(),
+                Err(e) => {
+                    error!(?event_id, "Failed to get unthreaded read receipts for event: {e}");
+                    IndexMap::new()
+                }
+            };
+
+        let main_thread_receipts =
+            match self.event_receipts(ReceiptType::Read, ReceiptThread::Main, event_id).await {
+                Ok(receipts) => receipts,
+                Err(e) => {
+                    error!(?event_id, "Failed to get main thread read receipts for event: {e}");
+                    Vec::new()
+                }
+            };
+
+        unthreaded_receipts.extend(main_thread_receipts);
```

Comment:
Should we try to deduplicate receipts here? Also, for my own curiosity: why aren't we including private read receipts here?

Reply:
We cannot deduplicate receipts here, we do not know which one is more recent yet. This is something we will do later in `maybe_update_read_receipt`.

We are not including private read receipts here because we can only have one: the one from our own user; and we already loaded it when building the timeline.